students/12th/soohyunlee/project_westagram/posting/views.py

```diff
@@ -0,0 +1,22 @@
+import json
+
+from django.http   import JsonResponse
+from django.views  import View
+from .models       import Posting
+from user.models   import User
+
+class PostUpView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        writer_user = User.objects.get(name = data['writer'])
+        if data['writer'] == writer_user.name:
+            Posting(
+                writer_id   = writer_user.id,
+                contents = data['contents'],
+                img_url  = data['img_url']
+            ).save()
+        return JsonResponse({"message":"SUCCESS"}, status=201)
+
+    def get(self, request):
+        posting_data = Posting.objects.values()
```

Comment:
`values()`메소드를 사용하면 그 클래스에 포함된 속성들만 모두 딕셔너리 형태로 보여주기 때문에 writer의 name 속성에 접근하기 어렵습니다.
`all()`이라는 메소드는 포함된 데이터를 객체로 반환하기 때문에 그 객체(Posting)의 속성(writer)이 또 하나의 다른 객체(User)라면, 그 객체(User)의 속성(name)에도 접근할 수 있습니다.
대신 values()와 달리 딕셔너리 모양으로 만들어주지 않기 때문에 우리가 따로 딕셔너리화 해주어야 합니다.
속성들에 매우 자유롭게 접근할 수 있기 때문에 values()는 거의 쓰지 않고 all()을 보다 더 많이 사용합니다 👍 

```suggestion
        posting_data = Posting.objects.all()
        posting_list = []
        for posting in posting_data:
            posting_list.append(
                {
                    'contents' : posting.contents,
                    'writer'      : posting.writer.name
                }
            )
```

Reply:
아 넵 ! 감사합니다 ~~!